src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -501,6 +501,11 @@ buffer_manager_allocate_buffer_for_thread (
 	EP_ASSERT(buffer_size > 0);
 	ep_return_null_if_nok(buffer_manager_try_reserve_buffer(buffer_manager, buffer_size));
 
+	// The sequence counter is exclusively mutated on this thread so this is a thread-local read.
+	sequence_number = ep_thread_session_state_get_volatile_sequence_number (thread_session_state);
+	new_buffer = ep_buffer_alloc (buffer_size, ep_thread_session_state_get_thread (thread_session_state), sequence_number);
+	ep_raise_error_if_nok (new_buffer != NULL);
+
 	// Allocating a buffer requires us to take the lock.
```

Comment:
This comment seems to indicate an allocation requires the lock. Is this no longer true or is the new allocation not the one the comment is referring to?

Reply:
I believe this comment is outdated. Changes in the .net6 timeframe made the reservation of the buffer size lockless and was moved out of the lock. Previously, this action required the lock and so the whole allocation was behind the lock. I'll update the comment.